src/smt/proof_post_processor.cpp

```diff
@@ -959,7 +961,7 @@ Node ProofPostprocessCallback::expandMacros(ProofRule id,
                           << std::endl;
     return sumBounds;
   }
-  else if (id == ProofRule::STRING_INFERENCE)
+  else if (id == ProofRule::MACRO_STRING_INFERENCE)
```

Comment:
If I understand correctly, the process here will add a theory inference trusted step when the elaboration fails, right?

Reply:
Yes.